cpu/stm32/periph/adc_f3.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2020 LAAS-CNRS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_stm32
+ * @ingroup     drivers_periph_adc
+ * @{
+ *
+ * @file
+ * @brief       Low-level ADC driver implementation
+ *
+ * @author      Hugues Larrive <hugues.larrive@laas.fr>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "mutex.h"
+#include "periph/adc.h"
+#include "periph_conf.h"
+#include "xtimer.h"
```

Comment:
You want to add

```
ifneq (,$(filter periph_adc,$(FEATURES_USED)))
  ifneq (,$(filter f3 l4,$(CPU_FAM)))
    USEMODULE += xtimer
  endif
endif
```

to `stm32/Makefile.dep`

Reply:
I think stm32 adc implementation would require a global clean-up so it's out of scope here.